test/extended/dr/common.go

```diff
@@ -649,26 +681,162 @@ EOF
+		  ssh -i $P_KEY -o StrictHostKeyChecking=no -q core@${i} sudo rm -rf /etc/kubernetes/manifests/etcd-pod.yaml 
+          echo "remove data dir on [$i]" 
+          ssh -i $P_KEY -o StrictHostKeyChecking=no -q core@${i} sudo rm -rf /var/lib/etcd
+          echo "restarting kubelet on [$i]"
+		  ssh -i $P_KEY -o StrictHostKeyChecking=no -q core@${i} sudo systemctl restart kubelet.service
+		done
+
+        TARGET_NODE_NAME=%s 
+        ssh -i $P_KEY -o StrictHostKeyChecking=no -q core@${TARGET_NODE_NAME} <<EOF
+         sudo rm -rf /home/core/backup
+         scp -o StrictHostKeyChecking=no -r core@%s:/home/core/backup .
+         sudo chown -R core /home/core/backup
+         SNAPSHOT=\$(ls -vd /home/core/backup/snapshot*.db | tail -1)
+         mv \$SNAPSHOT /home/core/backup/snapshot.db
+
+         sudo /usr/local/bin/cluster-restore.sh /home/core/backup
+         # this will cause the pod to disappear effectively, must be the last statement
+         sudo systemctl restart kubelet.service
+
+EOF`, sshKeyDance, strings.Join(nonRecoveryIps, " "), restoreInternalIp, backupInternalIp)
+
+	podSpec := applycorev1.PodSpec().WithHostNetwork(true).WithRestartPolicy(corev1.RestartPolicyOnFailure)
+	podSpec.Containers = []applycorev1.ContainerApplyConfiguration{
+		*applycorev1.Container().
+			WithName("cluster-restore").
+			WithSecurityContext(applycorev1.SecurityContext().WithPrivileged(true)).
+			WithImage(image.ShellImage()).
+			WithVolumeMounts(
+				applycorev1.VolumeMount().WithName("keys").WithMountPath(sshPath),
+			).
+			WithCommand("/bin/bash", "-c", restoreScript),
+	}
+
+	podSpec.NodeSelector = map[string]string{"kubernetes.io/hostname": restoreNode.Labels["kubernetes.io/hostname"]}
+	podSpec.Tolerations = []applycorev1.TolerationApplyConfiguration{
+		*applycorev1.Toleration().WithKey("node-role.kubernetes.io/master").WithOperator(corev1.TolerationOpExists).WithEffect(corev1.TaintEffectNoSchedule),
+	}
+
+	podSpec.Volumes = []applycorev1.VolumeApplyConfiguration{
+		*applycorev1.Volume().WithName("keys").WithSecret(applycorev1.SecretVolumeSource().WithSecretName("dr-ssh")),
+	}
+
+	pod := applycorev1.Pod(name, namespace).WithSpec(podSpec)
+	// we only run the pod and not wait for it, as it will not be tracked after the control plane comes back
+	return runPod(oc, pod)
+}
+
+// TODO(thomas): according to Dan: technically you don't need to wipe the OVN databases,
+// just restarting ovnkube-master after making sure etcd/apiserver are serving the new information should be OK.
+func runOVNRepairCommands(oc *exutil.CLI, restoreNode *corev1.Node, nonRecoveryNodes []*corev1.Node) error {
+	const name = "etcd-ovn-repair-pod"
+	const namespace = "openshift-etcd"
+	framework.Logf("running ovn-repair script on node: %v", restoreNode.Name)
+
+	var nodeIPs []string
+	for _, n := range nonRecoveryNodes {
+		nodeIPs = append(nodeIPs, internalIP(n))
+	}
+	// adding this node last is important for kubelet restarts, as this node runs below pod
```

Comment:
Nit: For my own clarification, this is so the ovn-repair pod can restart its underlying kubelet at the very end right?
If so, then maybe a slight clarification:
```suggestion
	// adding the restore node last is important for kubelet restarts, as the ovn-repair pod runs on top of that node
```

Also what happens if this isn't the case? The ovn-repair pod goes away after it restarts its own kubelet?

Reply:
applied. 

> Also what happens if this isn't the case? The ovn-repair pod goes away after it restarts its own kubelet?

correct. I believe we can remove the pod soon. Dan confirmed it's enough to just remove the ovnkube master pods. I'm just waiting for Allen's changes, so we can verify the watch resume more easily.